### PR TITLE
Bugfixes and small stuff

### LINKED ISSUE #151: clicking the refresh button too fast can cause posts to load multiple times
this has been known for a while i just haven't ever gotten around to fixing it

### LINKED ISSUE #152: make line limit on quoted posts smaller

### LINKED ISSUE #153: add that unsaved changes alert when leaving the settings page with unsaved changes

### LINKED ISSUE #155: put something in title to signify unread notifications/messages
could just be added on to the notification refresh function in base_footer, and then just make it add `(*)` or something to the title if there are new notifications. would need to make sure that this only shows once though

### LINKED ISSUE #156: badges arent shown on messages

### PR DESCRIPTION
**Pull request by trinkey**
## What does this do?
refreshing the timeline quickly no longer shows duplicate posts, quoted posts now have a smaller line limit, you now get prompted when you leave the settings page with unsaved changes, the title now signifies if you have unread notifications, and badges are now shown in message pages.

## Issues completed:
- fixes #151
- fixes #152
- fixes #153
- fixes #155
- fixes #156

---
### Checklist for Reviewers
- [x] Version has been incremented (if appropriate)
- [x] BACKEND.md has been updated for any new functions
- [x] Code is formatted good enough (ask @trinkey)
- [x] Any new changes have been tested thoroughly (including api calls that wouldn't normally be possible using inspect element or <localhost:port or url>/api/docs)
- [x] Features that break easily have been tested (mostly viewing pages when not logged in)

### CODE REVIEW

**Comment by trinkey** [comment]
actually #156 isnt fully done yet it's not showing on /messages, only /m/...

File: smiggins/static/js/base.js [hunk 1]
```diff
@@ -302,7 +302,7 @@ const icons = {
...
-  lock     : '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><title>Private Account</title><path d="M144 128v64h160v-64c0-44.2-35.8-80-80-80s-80 35.8-80 80zm-48 64v-64C96 57.3 153.3 0 224 0s128 57.3 128 128v64h32c35.3 0 64 28.7 64 64v192c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V256c0-35.3 28.7-64 64-64h32zm-48 64v192c0 8.8 7.2 16 16 16h320c8.8 0 16-7.2 16-16V256c0-8.8-7.2-16-16-16H64c-8.8 0-16 7.2-16 16z"/></svg>',
+  lock     : '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><title>Private Account</title><path transform="scale(0.85) translate(10 70)" d="M144 128v64h160v-64c0-44.2-35.8-80-80-80s-80 35.8-80 80zm-48 64v-64C96 57.3 153.3 0 224 0s128 57.3 128 128v64h32c35.3 0 64 28.7 64 64v192c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V256c0-35.3 28.7-64 64-64h32zm-48 64v192c0 8.8 7.2 16 16 16h320c8.8 0 16-7.2 16-16V256c0-8.8-7.2-16-16-16H64c-8.8 0-16 7.2-16 16z"/></svg>',
```
Comment (DiamondTaco) [inline smiggins/static/js/base.js:305]: fix bug here bad commit should add another 0

**Review by DiamondTaco — CHANGES_REQUESTED** [review]
lgtm

**Review by DiamondTaco — APPROVED** [review]
ew java script

**Pull request merged**